### PR TITLE
Improve security of systemd service rest-server.service by restricting network access

### PR DESCRIPTION
**Pull request by eriksjolund**
What is the purpose of this change? What does it change?
--------------------------------------------------------

Improve security of rest-server.service by restricting network access.

This patch improves the overall security assessment score given by `systemd-analyze security rest-server.service` from "1.3 OK" to "0.6 SAFE" (when using systemd-analyze version 253)

* Remove `AF_INET AF_INET6` from RestrictAddressFamilies. Sockets originating from socket activation are not affected by the systemd directive RestrictAddressFamilies. See [systemd.exec](https://www.freedesktop.org/software/systemd/man/systemd.exec.html) man page.

* Add `PrivateNetwork=yes` as recommended for socket-activated services in the [systemd.socket](https://www.freedesktop.org/software/systemd/man/systemd.socket.html) man page.

* Add dependency on rest-server.socket

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

Yes, in the forum:

https://forum.restic.net/t/using-none-instead-of-af-inet-af-inet6-for-restrictaddressfamilies-in-systemd-unit-rest-server-service/6448

Checklist
---------

- [x] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [ ] I have added tests for all changes in this PR
- [ ] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/rest-server/blob/master/changelog/TEMPLATE))
- [ ] I have run `gofmt` on the code in all commits
- [ ] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/rest-server/commits/master)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: examples/systemd/rest-server.service [hunk 1]
```diff
@@ -37,6 +34,13 @@ CapabilityBoundingSet=
 LockPersonality=true
 MemoryDenyWriteExecute=true
 NoNewPrivileges=yes
+
+# See the paragraph "All network sockets allocated ..." in the systemd.socket man page
```
Comment (MichaelEischer) [inline examples/systemd/rest-server.service:38]: I'd prefer to just explain why we use `PrivateNetwork` here:
```
As the listen socket is created by systemd via the rest-server.socket unit, it is no longer necessary for rest-server to have access to the host network namespace.
```

File: changelog/unreleased/pull-246 [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Enhancement: Enable socket activation by default and add network access restrictions in example systemd unit rest-server.service
```
Comment (MichaelEischer) [inline changelog/unreleased/pull-246:1]: We usually don't add changelog entries for documentation updates. Either way, this changelog entry is far too long. So I'd suggest to remove it.

File: examples/systemd/rest-server.service [hunk 3]
```diff
@@ -15,10 +14,8 @@ ExecStart=/usr/local/bin/rest-server --path /path/to/backups
 Restart=always
 RestartSec=5
 
-# The following options are available (in systemd v247) to restrict the
-# actions of the rest-server.
-
-# As a whole, the purpose of these are to provide an additional layer of
+# The following options restricts the actions of the rest-server.
```
Comment (MichaelEischer) [inline examples/systemd/rest-server.service:17]: The systemd version number was added some time ago as some of the used options are not available in older systemd versions. Thus, I'd prefer to keep it.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]
Thanks for opening a PR! I like the changes in general, but I think some comments and in particular the changelog are too verbose:

File: examples/systemd/rest-server.service (hunk 1 above)
Comment (eriksjolund) [reply examples/systemd/rest-server.service]: Done

File: examples/systemd/rest-server.service (hunk 3 above)
Comment (eriksjolund) [reply examples/systemd/rest-server.service]: Done

File: changelog/unreleased/pull-246 (hunk 2 above)
Comment (eriksjolund) [reply changelog/unreleased/pull-246]: Done

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**